autoscheduler/scraper/management/commands/scrape_courses.py

```diff
@@ -1,6 +1,133 @@
+    meeting_end_time = convert_meeting_time(meetings_data['meetingTime']['endTime'])
+    meeting_class_type = meetings_data['meetingTime']['meetingType']
+    #creates and saves meeting object
+    meeting_object = Meeting(id=meeting_id, building=meeting_building,
+                             meeting_days=meeting_class_days,
+                             start_time=meeting_start_time, end_time=meeting_end_time,
+                             meeting_type=meeting_class_type, section=section_object)
+    meeting_object.save()
+
+def parse_instructor(course_data):
+    """called from parse course. puts instructor data in database"""
+    #gets values to create object
+    for i in course_data['faculty']:
+        instructor_id = i['displayName']#primary key. name is currently being used
+        instructor_email = i['emailAddress']
+        #creates and saves instructor object
+        instructor_object = Instructor(
+            id=instructor_id, email_address=instructor_email)
+        instructor_object.save()
+        return instructor_object
+
+def parse_course(course_data):
+    """puts course data in database. calls parse instructor and section"""
+    course_dept = course_data['subject']
+    course_number = course_data['courseNumber']
+    course_title = course_data['courseTitle']
+    course_term_code = course_data['term']
+    course_id = generate_course_id(
+        course_dept, course_number, course_term_code)#primary key
+    course_credit_hours = course_data['creditHourLow']
+    #creates and saves course object
+    course_object = Course(
+        id=course_id, dept=course_dept, course_num=course_number, title=course_title,
+        credit_hours=course_credit_hours, term=course_term_code)
+    course_object.save()
+    #calls other parse functions
+    instructor_object = parse_instructor(course_data)
+    parse_section(course_data, instructor_object)
+
+def get_department_names(term_code) -> List[str]:
+    """queries database for list of all departments"""
+    return [dept.code for dept in Department.objects.filter(term=term_code)]
 
 class Command(base.BaseCommand):
-    "Iterates through all of the courses and loads them into the database"
-    def handle(self, *args, **kwargs):
-        pass
\ No newline at end of file
+    " Gets course information from banner and adds it to the database "
+
+    def add_arguments(self, parser):
+        parser.add_argument('--term', type=str, required=False, default="201931")
+
+    def handle(self, *args, **options):
+        banner = BannerRequests(options['term'])
+        loop = asyncio.get_event_loop()
+        # the next line should be used in production, but we should only use two or three
+        # departments when testing to make the code run way faster
+        # once department scraping is done, we can replace get_department_names
+        # with just getting each of the departments for the term from our database
+        depts = get_department_names(banner)
```

Comment:
Correct me if I'm wrong, but in order to get this to work, I had to change this line to `depts = get_department_names(options['term'])` Since the argument to `get_department_names` should be a term code, not a `BannerRequests` object.